tracer/src/Datadog.Trace/Util/Http/HttpRequestUtils.cs

```diff
@@ -0,0 +1,27 @@
+// <copyright file="HttpRequestUtils.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.Util.Http
+{
+    internal class HttpRequestUtils
+    {
+        private const string NoHostSpecified = "UNKNOWN_HOST";
+
+        internal static string GetUrl(string scheme, string host, string pathBase, string path, Func<string> getQueryString, ImmutableTracerSettings tracerSettings = null)
+        {
+            if (tracerSettings != null && tracerSettings.EnableQueryStringReporting)
+            {
+                var queryStringObfuscator = QueryStringObfuscator.Instance(tracerSettings.ObfuscationQueryStringRegexTimeout, tracerSettings.ObfuscationQueryStringRegex);
```

Comment:
Should we include QueryStringObfuscator instance in TracerManager? It doesn't feel right to pass the settings up to here.

Reply:
if `QueryStringObfuscator `is a singleton, how could the regex be re created everytime?

Reply:
Well, it can't, which is the point 😄 `TracerSettings` won't change every request, they'll only change when `Tracer.Configure()` is called, so we can create it once then and avoid the allocations 🙂

Reply:
I meant it's already a singleton as it is now.. but it's true that if tracer get reconfigured it wont be reinstantiated with the changes 😅 so will make it part of the `TracerManager` instance

Reply:
changed in 2016345d3257fff2e9bbd8f4321448602bd523eb